finding-distinct/ksql/README.md

```diff
@@ -0,0 +1,55 @@
+# How to filter duplicate events from a Kafka topic with ksqlDB
+
+Consider a topic with events that represent clicks on a website. Each event contains an IP address, a URL, and a timestamp.
+In this tutorial, we'll use ksqlDB to deduplicate these click events.
+
+## Setup
+
+Let's start with a stream representing a user click events:
+
+```sql
+CREATE STREAM CLICKS (IP_ADDRESS VARCHAR, URL VARCHAR)
+    WITH (KAFKA_TOPIC = 'CLICKS',
+          FORMAT = 'JSON',
+          PARTITIONS=1);
+```
+
+Then create a table to collect the click events:
+
+```sql
+CREATE TABLE DETECTED_CLICKS AS
+    SELECT
+        IP_ADDRESS AS KEY1,
+        URL AS KEY2,
+        AS_VALUE(IP_ADDRESS) AS IP_ADDRESS,
+        COUNT(IP_ADDRESS) as IP_COUNT,
+        AS_VALUE(URL) AS URL,
+        FORMAT_TIMESTAMP(FROM_UNIXTIME(EARLIEST_BY_OFFSET(ROWTIME)), 'yyyy-MM-dd HH:mm:ss.SSS') AS TIMESTAMP
+    FROM CLICKS WINDOW TUMBLING (SIZE 2 MINUTES, RETENTION 1000 DAYS)
+    GROUP BY IP_ADDRESS, URL;
+```
+
+As we’re grouping by ip-address and url, these columns will become part of the primary key of the table. Primary key columns are stored in the Kafka message’s key. As we’ll need them in the value later, we use AS_VALUE to copy the columns into the value and set their name. To avoid the value column names clashing with the key columns, we add aliases to rename the key columns.
```

Comment:
```suggestion
As we’re grouping by ip-address and url, these columns will become part of the primary key of the table. Primary key columns are stored in the Kafka message’s key. As we’ll need them in the value later, we use `AS_VALUE` to copy the columns into the value and set their name. To avoid the value column names clashing with the key columns, we add aliases to rename the key columns.
```